### PR TITLE
Documentation for Native Hash Table

### PR DESCRIPTION
**Pull request by cshung**
@dotnet/crossgen-contrib 
@jkotas

### CODE REVIEW

**Comment by trylek** [comment]
Thanks Andrew for providing the initial description. While I'm not as sadistic as to nag you to provide a picture, it should be relatively straightforward to at least include a table outlining the contents of the native hashtable - I suspect that, with just three columns (section, size, purpose) and three rows (header - or maybe multiple lines for the individual header fields / buckets / values) you should be able to demonstrate the hashtable format in a way that's much easier to grasp.

**Comment by trylek** [comment]
Beautiful, thank you so much!

File: Documentation/botr/readytorun-format.md [hunk 1]
```diff
@@ -460,7 +460,50 @@ the first byte of the encoding specify the number of following bytes as follows:
...
+
+To make look up fast, the number of buckets is always a power of 2. The table is simply a sequence of `(1 + number of buckets)` cells, for the first `(number of buckets)` cells, its stores the offset of the bucket list from the beginning of the whole native hash table. The last cell stores the offset to the end of the buckets.
+
+Each bucket is a sequence of entries. An entry has a hash code and an offset to the object stored. The entries are sorted by hash code.
+
+Physically, the header is a single byte. The most significant six bits is used to store the number of buckets in its base-2 logarithm. The remaining two bits are used for storing the entry size, as explained below:
+
+Because the offsets to the bucket lists are often small numbers, the table cells are variable sized. 
+It could be either 1 byte, 2 bytes or 4 bytes. The three cases are described with two bits. `00` means it is one byte, `01` means it is two bytes and `02` means it is four bytes.
```
Comment (jakobbotsch) [inline Documentation/botr/readytorun-format.md]: `02`?

Comment (cshung) [reply Documentation/botr/readytorun-format.md]: Thanks for the catch, of course, I meant `10`

**Pull request merged**